test/oxo.js

```diff
@@ -0,0 +1,110 @@
+const Reverter = require('./helpers/reverter');
+const Asserts = require('./helpers/asserts');
+const OXO = artifacts.require('./OXO.sol');
+
+contract('OXO Game', function (accounts) {
+  const reverter = new Reverter(web3);
+  afterEach('revert', reverter.revert);
+
+  const asserts = Asserts(assert);
+  const Player1 = accounts[1];
+  const Player2 = accounts[2];
+  let oxoGame;
+
+  before('setup', () => {
+    return OXO.deployed()
+      .then(instance => (oxoGame = instance))
+      .then(reverter.snapshot);
+  });
+
+  it('should allow to create game', async () => {
+    await oxoGame.createGame({from: Player1, value: 100});
+    var result = await oxoGame.createGame.call({from: Player1, value: 100});
+    assert.equal(result, 1);
+  });
+
+  it('should allow to join game', async () => {
+    await oxoGame.createGame({from: Player1, value: 100});
+    assert.isTrue(await oxoGame.joinGame.call(0, {from: Player2, value: 100}));
+  });
+
+  it('should not allow to join yourself', async () => {
+    await oxoGame.createGame({from: Player1, value: 100});
+    await asserts.throws(oxoGame.joinGame.call(0, {from: Player1, value: 100}));
+  });
+
+  it('should not allow to join with differend bet size', async () => {
+    await oxoGame.createGame({from: Player1, value: 100});
+    await asserts.throws(oxoGame.joinGame.call(0, {from: Player1, value: 50}));
+  });
+
+  it('should allow to join only game with NEW status', async () => {
+    await oxoGame.createGame({from: Player1, value: 100});
+    await oxoGame.joinGame(0, {from: Player2, value: 100});
+    await asserts.throws(oxoGame.joinGame.call(0, {from: Player2, value: 100}));
+  });
+
+  it('should allow to make move', async () => {
+    await oxoGame.createGame({from: Player1, value: 100});
+    await oxoGame.joinGame(0, {from: Player2, value: 100});
+    asserts.isTrue(await oxoGame.makeMove.call(0, 1, {from: Player1}));
+  });
+
+  it('should not allow to make move second time', async () => {
+    await oxoGame.createGame({from: Player1, value: 100});
+    await oxoGame.joinGame(0, {from: Player2, value: 100});
+    await oxoGame.makeMove(0, 1, {from: Player1});
+    await asserts.throws(oxoGame.makeMove.call(0, 1, {from: Player1}));
```

Comment:
Maybe it would be better if you check here another fieldNumber. In order to check that you can't make three moves in a row and get a win combination before second player even make his first move.

Reply:
Yep, I will change the fieldNumber for second move to look more logical, but it still correct test